R/ff_analyze.R

```diff
@@ -1,130 +1,243 @@
+      previous_data <- read.csv(csv_filename)
+      previous_data$X <- NULL
+      write.csv(rbind(previous_data, polygons_dataframe), csv_filename)
     } else {
-      if (!file.exists(csvfile) & append & verbose) {
-        ff_cat("the given file does not exist, while append was set to TRUE", color = "yellow", verbose = verbose)
+      if (!file.exists(csv_filename) && append && verbose) {
+        ff_cat("the given file does not exist, while append was set to TRUE",
+          color = "yellow", verbose = verbose
+        )
       }
-      write.csv(pol_df, csvfile)
+      write.csv(polygons_dataframe, csv_filename)
+    }
+  }
+
+  return(polygons_dataframe)
+}
+
+#' Calculate Scores
+#'
+#' This function calculates scores based on predictions, ground truth, and optional parameters.
+#' It processes raster data and compares predictions against ground truth, optionally using
+#' a forest mask to restrict the analysis area.
+#'
+#' @param predictions A SpatRaster object or path to raster file representing the predictions
+#' @param groundtruth A SpatRaster object or path to raster file representing the ground truth
+#' @param forest_mask Optional SpatRaster object or path to raster file representing the forest mask
+#' @param csv_filename Optional path to CSV file for writing results
+#' @param country Character string containing ISO3 code for filtering analysis polygons
+#' @param append Logical indicating whether to append results to existing CSV file
+#' @param analysis_polygons Optional SpatVector object or path to vector file for analysis areas
+#' @param remove_empty Logical indicating whether to remove empty records
+#' @param date Character string in YYYY-MM-DD format
+#' @param tile Character string in format AA{N-S}_BBB{W-E}
+#' @param method Character string indicating the analysis method
+#' @param add_wkt Logical indicating whether to add WKT geometry to output
+#' @param verbose Logical indicating whether to print progress messages
+#'
+#' @return A SpatVector object containing calculated scores for each polygon
+#'
+#' @export
+ff_analyze <- function(predictions, groundtruth, forest_mask = NULL, csv_filename = NULL,
+                       country = NULL, append = TRUE, analysis_polygons = NULL,
+                       remove_empty = TRUE, date = NULL, tile = NULL, method = NA,
+                       add_wkt = FALSE, verbose = FALSE) {
+  # Get date if not provided
+  if (is.null(date)) {
+    date <- get_date_from_files(predictions, groundtruth)
+  }
+
+  # Validate and load input data
+  data <- validate_and_load_data(predictions, groundtruth, forest_mask)
+  predictions <- data$predictions
+  groundtruth <- data$groundtruth
+  forest_mask <- data$forest_mask
+
+  ff_cat("rasters loaded", verbose = verbose)
+
+  # Create crosstable raster
+  if (!is.null(forest_mask)) {
+    ff_cat("using forest mask", verbose = verbose)
+    crosstable_raster <- (2 * groundtruth + predictions) * (forest_mask > 0)
```

Comment:
Are you sure you moved the crosstable raster here? It might affect the logic since previously it was checked before the forestmask got any value

Reply:
was it? it should definitely check if a forest mask was available before creating the crosstable so this is correct